grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/config/DatastoreServiceMethodInvokingFactoryBean.groovy

```diff
@@ -22,7 +22,10 @@ class DatastoreServiceMethodInvokingFactoryBean extends MethodInvokingFactoryBea
 
     @Override
     Class<?> getObjectType() {
-        arguments[0] as Class<?>
+        if (arguments) {
```

Comment:
I think the check should be `if (arguments != null && arguments.size() == 1 )`. Also, instead of returning `null` maybe we should call `super.getObjectType()`.